pkg/bootstrap/platform/gcp.go

```diff
@@ -148,26 +172,41 @@ func (e *gcpEnv) Metadata() map[string]string {
 	if e.metadata != nil {
 		return e.metadata
 	}
-	envPid, envNPid, envCN, envLoc := parseGCPMetadata()
-	if envPid != "" {
+	envPid, envNPid, envCN, envLoc, err := parseGCPMetadata()
+
+	// If no error is returned, then we have a valid GCP environment.
+	// Else we query the API to discover the needed information
+	if err == nil {
```

Comment:
i think we want per-value checking, with a single available check.

```
md[GCPProject] = envPid
...
if isMetadataEndpointAccessibleFn() {
  // fill in empty values if possible
  if md[GCPProject] == "" {...}
  ...
}

Thoughts?

Reply:
I updated this a little to follow your suggestion, but assigning empty values to the map causes some unit tests to fail ( because of the method `constructGKEClusterURL` that expects entries to either be valid or not present), we can update the logic in that instance, but maybe there are other areas of the code that expect the same. (We can clean up all those areas if you think it's worthwhile)